py/src/pathfinder_worker/call.py

```diff
@@ -342,14 +343,23 @@ def do_loop(connection, input_gen, output_file):
         except marshmallow.exceptions.MarshmallowError as exc:
             logger.error(f"Failed to parse command: {exc}")
             out = {"status": "error", "kind": "INVALID_INPUT"}
-        except WebFriendlyException as exc:
-            if str(exc.code) == "StarknetErrorCode.UNINITIALIZED_CONTRACT":
+        except StarkException as exc:
+            if exc.code == StarknetErrorCode.UNINITIALIZED_CONTRACT:
                 out = {"status": "error", "kind": "NO_SUCH_CONTRACT"}
-            elif str(exc.code) == "StarknetErrorCode.ENTRY_POINT_NOT_FOUND_IN_CONTRACT":
+            elif exc.code == StarknetErrorCode.ENTRY_POINT_NOT_FOUND_IN_CONTRACT:
```

Comment:
Is there value in having message data for these other two variants? Maybe it tells you which contract is unitialized or which entry point was not found?

Reply:
That'd be a much more intrusive change. Those are explicit errors returned in our response and not generic failures. Our Rust code cannot handle arbitrary messages for those cases (see https://github.com/eqlabs/pathfinder/blob/913c1de9640d3f405288fbf61306d9867a68cb66/crates/rpc/src/cairo/ext_py/de.rs#L105 for details).